var/spack/repos/builtin/packages/libogg/package.py

```diff
@@ -24,3 +28,35 @@ class Libogg(AutotoolsPackage):
         sha256="0f4d289aecb3d5f7329d51f1a72ab10c04c336b25481a40d6d841120721be485",
         when="@1.3.4 platform=darwin",
     )
+    build_system(
+        conditional("cmake", when="@1.3.4:"),
+        conditional("generic", when="@1.3.2 platform=windows"),
+        "autotools",
+        default="autotools",
+    )
+
+
+class GenericBuilder(GenericBuilder):
+    phases = ["build", "install"]
+
+    def is_64bit(self):
+        return platform.machine().endswith("64")
```

Comment:
Like https://github.com/spack/spack/pull/35095, can this analyze the spec to determine 64-bit support?

Other than this the PR looks good.

Reply:
Oh huh I thought I already went back and did this to all my prs, guess I missed it here, good catch!

Reply:
I'll also make a new PR converting perl to this style